usd/src/usd_parser/USDMaterial.cc

```diff
@@ -39,18 +39,41 @@ namespace sdf
   /// \param[in] _dest The destination for the copy of _ori
   /// \return A list of UsdErrors. An empty list means no errors occurred when
   /// copying _ori to _dest
-  UsdErrors copyFile(const std::string &_ori, const std::string &_dest)
+  UsdErrors copyFile(const std::string &_ori, std::string &_dest)
   {
     UsdErrors errors;
     if (ignition::common::exists(_ori))
     {
-      // If the file exists then we should remove it and we will copy the
-      // file below
-      if (ignition::common::exists(_dest))
+      // If the file exists then we append a number suffix to the destination
+      // file
+      // For example:
+      // /bar/foo.extension
+      // /bar/foo_X.extension
```

Comment:
It would be nice to put all of the code for appending a numeric suffex to the destination file in an `if (ignition::common::exists(_dest))` block. That way, we don't do the preliminary work of saving the parent path, file extension, and file name without extension unless we absolutely have to.